silence/silence_test.go

```diff
@@ -109,6 +109,119 @@ func TestSilencesGC(t *testing.T) {
 	require.Equal(t, want, s.st)
 }
 
+func TestSilenceGCOverTime(t *testing.T) {
+	type silenceEntry struct {
+		s                    *pb.Silence
+		expectPresentAfterGc bool
+	}
+	type testCase struct {
+		initialState    []silenceEntry
+		updates         []silenceEntry
+		expectedGCCount int
+	}
+
+	c := clock.NewMock()
+	now := c.Now().UTC()
+
+	newSilence := func(id string, exp time.Time) *pb.Silence {
+		return &pb.Silence{
+			Id:       id,
+			StartsAt: now.Add(-time.Second),
+			EndsAt:   exp,
+			Matchers: []*pb.Matcher{
+				{Name: "foo", Type: pb.Matcher_REGEXP, Pattern: "bar"},
+			}}
+	}
+
+	cases := map[string]testCase{
+		"gc does not clean active silences": {
+			initialState: []silenceEntry{
+				{s: newSilence("1", now), expectPresentAfterGc: false},
+				{s: newSilence("2", now.Add(-time.Second)), expectPresentAfterGc: false},
+				{s: newSilence("3", now.Add(time.Second)), expectPresentAfterGc: true},
+			},
+		},
+		"silences added with Set are handled correctly": {
+			initialState: []silenceEntry{
+				{s: newSilence("1", now), expectPresentAfterGc: false},
+			},
+			updates: []silenceEntry{
+				{s: newSilence("", now.Add(time.Second)), expectPresentAfterGc: true},
+				{s: newSilence("", now.Add(-time.Second)), expectPresentAfterGc: false},
+			},
+		},
+		"silence update does not leak state": {
+			initialState: []silenceEntry{
+				{s: newSilence("1", now), expectPresentAfterGc: false},
+			},
+			updates: []silenceEntry{
+				{s: newSilence("1", now.Add(time.Second)), expectPresentAfterGc: true},
+			},
+		},
+	}
+
+	for name, tc := range cases {
+		t.Run(name, func(t *testing.T) {
+			silences, err := New(Options{})
```

Comment:
Please just move the `require` after the `New` call:

```suggestion
			silences, err := New(Options{})
			require.NoError(t, err)
```